### PR TITLE
Feature/landing page

### PR DESCRIPTION
**Pull request by AtanasKrondev**
#13 
Landing page design close to the original one.
Carousels animations and autoplay to be done.

### CODE REVIEW

**Comment by mradkov** [comment]
> Carousels animations and autoplay to be done.

Are you going to add these in this PR or should I review and merge this one and expect another PR with the animations?

**Comment by AtanasKrondev** [comment]
> > Carousels animations and autoplay to be done.
> 
> Are you going to add these in this PR or should I review and merge this one and expect another PR with the animations?

I will add it in this PR so no need to review it twice.

**Comment by mradkov** [comment]
Please mark when the PR is ready for review :)

**Comment by AtanasKrondev** [comment]
> Please mark when the PR is ready for review :)

It can be reviewed :)

File: src/components/landing/Projects.vue [hunk 1]
```diff
@@ -450,29 +480,39 @@ export default {
...
         this.activeSlide = this.activeSlide + 1;
       }
     },
   },
+  computed: {
+    direction: function () {
+      if (this.prevSlide > this.activeSlide) {
+        return "right";
+      } else return "left";
+    },
```
Comment (mradkov) [inline src/components/landing/Projects.vue:504]:
```suggestion
    direction: () => {
      return this.prevSlide > this.activeSlide ? 'right' : 'left';
    },
```

File: src/components/landing/Projects.vue [hunk 2]
```diff
@@ -450,29 +480,39 @@ export default {
...
       immediate: true,
       handler(newSlide, oldSlide) {
         clearInterval(interval);
         interval = setInterval(() => {
-          this.setActive(newSlide + 1);
-          if (this.activeSlide > 10) {
+          if (newSlide + 1 > 10) {
             this.setActive(1);
+          } else {
+            this.setActive(newSlide + 1);
```
Comment (mradkov) [inline src/components/landing/Projects.vue]:
```suggestion
          this.setActive(newSlide + 1 > 10 ? 1 : newSlide + 1);
```

Can be simplified like this. Also, where is this `10` coming from?

Comment (AtanasKrondev) [reply src/components/landing/Projects.vue]: Indeed. `10` is the hardcoded length of the slides.

File: src/components/landing/Projects.vue [hunk 3]
```diff
@@ -5,549 +5,459 @@
...
+              <li @click="setActive(2)" :class="{active: activeSlide === 2}"></li>
+              <li @click="setActive(3)" :class="{active: activeSlide === 3}"></li>
+              <li @click="setActive(4)" :class="{active: activeSlide === 4}"></li>
+              <li @click="setActive(5)" :class="{active: activeSlide === 5}"></li>
+              <li @click="setActive(6)" :class="{active: activeSlide === 6}"></li>
+              <li @click="setActive(7)" :class="{active: activeSlide === 7}"></li>
+              <li @click="setActive(8)" :class="{active: activeSlide === 8}"></li>
+              <li @click="setActive(9)" :class="{active: activeSlide === 9}"></li>
+              <li @click="setActive(10)" :class="{active: activeSlide === 10}"></li>
+            </ol>
```
Comment (mradkov) [inline src/components/landing/Projects.vue:450]:
```suggestion
            <ol class="carousel-indicators">
              <li 
                v-for="(item, index) in slides"
                :key="`slide-${index}`"
                @click="setActive(index)"
                :class="{ active: activeSlide === index }"
              ></li>
            </ol>
```

File: src/components/landing/Projects.vue [hunk 4]
```diff
@@ -556,5 +466,105 @@
 </template>
 
 <script>
-export default {}
+let interval;
+export default {
+  data() {
+    return {
+      activeSlide: 1,
+      prevSlide: 1,
```
Comment (mradkov) [inline src/components/landing/Projects.vue:474]:
```suggestion
      prevSlide: 1,
      slides: new Array(10),
```

**Comment by mradkov** [comment]
I think it will be best if we can extract all of the projects in a data structure (JSON or sth else) in a separate file, which we can then load and iterate over and render the necessary DOM elements. 🤔

**Comment by AtanasKrondev** [comment]
> I think it will be best if we can extract all of the projects in a data structure (JSON or sth else) in a separate file, which we can then load and iterate over and render the necessary DOM elements. Also, it will be easier for adding/modify/remove projects from that list. 🤔
> 
> That way we dont need to specify a specific count of items and can dynamically render everything needed based on the `lenght` of the items.
> 
> e.g.
> 
> ```json
> {
>   "projects": [
>     {
>       "title": "Example project",
>       "features": [
>         "users",
>         "admin panel",
>         "etc..."
>       ],
>       "technologies": [
>         "ethereum",
>         "nodejs",
>         "etc"
>       ]
>     },
>     {},
>     {}
>   ]
> }
> ```

Can you take another look :)

File: src/components/landing/Home.vue [hunk 5]
```diff
@@ -25,6 +21,25 @@
...
-export default {}
-</script>
+<style scoped>
+.home-bg {
+  height: 100%;
+  min-height: 100vh;
+}
+
+.hack-home-content {
+  /* position: absolute; */
```
Comment (mradkov) [inline src/components/landing/Home.vue:31]:
```suggestion
```

File: src/components/landing/Home.vue [hunk 6]
```diff
@@ -25,6 +21,25 @@
...
+  margin-top: 10%;
+}
+
+.hack-home-title {
+  white-space: wrap;
+  color: #9013fe;
+
+  /* text-shadow: -1px 0 #9013FE, 0 1px #9013FE, 1px 0 #9013FE, 0 -1px #9013FE; */
```
Comment (mradkov) [inline src/components/landing/Home.vue:39]:
```suggestion
```

File: src/components/landing/Home.vue [hunk 7]
```diff
@@ -25,6 +21,25 @@
...
+}
+
+.hack-home-tilte-p {
+  /* text-shadow: -1px 0 #9013FE, 0 1px #9013FE, 1px 0 #9013FE, 0 -1px #9013FE; */
```
Comment (mradkov) [inline src/components/landing/Home.vue:43]:
```suggestion
```

**Pull request merged**